scheduler/task.go

```diff
@@ -295,8 +295,11 @@ func (t *task) stream() {
 			}
 			select {
 			case <-t.killChan:
+				t.Lock()
 				t.state = core.TaskStopped
-				break
+				t.Unlock()
+				done = true
```

Comment:
It works exactly in the same way as for regular spin() - so it happens here: https://github.com/intelsdi-x/snap/blob/master/scheduler/task.go#L360